src/libs/ReportUtils.ts

```diff
@@ -216,278 +237,208 @@ function isCanceledTaskReport(report = {}, parentReportAction = {}) {
  */
-function isOpenTaskReport(report, parentReportAction = {}) {
-    return isTaskReport(report) && !isCanceledTaskReport(report, parentReportAction) && report.stateNum === CONST.REPORT.STATE_NUM.OPEN && report.statusNum === CONST.REPORT.STATUS.OPEN;
+function isOpenTaskReport(report: OnyxEntry<Report>, parentReportAction?: OnyxEntry<ReportAction>): boolean {
+    return isTaskReport(report) && !isCanceledTaskReport(report, parentReportAction) && report?.stateNum === CONST.REPORT.STATE_NUM.OPEN && report?.statusNum === CONST.REPORT.STATUS.OPEN;
 }
 
 /**
  * Checks if a report is a completed task report.
- *
- * @param {Object} report
- * @returns {Boolean}
  */
-function isCompletedTaskReport(report) {
-    return isTaskReport(report) && report.stateNum === CONST.REPORT.STATE_NUM.SUBMITTED && report.statusNum === CONST.REPORT.STATUS.APPROVED;
+function isCompletedTaskReport(report: OnyxEntry<Report>) {
+    return isTaskReport(report) && report?.stateNum === CONST.REPORT.STATE_NUM.SUBMITTED && report?.statusNum === CONST.REPORT.STATUS.APPROVED;
 }
 
 /**
  * Checks if the current user is the manager of the supplied report
- *
- * @param {Object} report
- * @returns {Boolean}
  */
-function isReportManager(report) {
-    return report && report.managerID === currentUserAccountID;
+function isReportManager(report: OnyxEntry<Report>): boolean {
+    return report?.managerID === currentUserAccountID;
 }
 
 /**
  * Checks if the supplied report has been approved
- *
- * @param {Object} report
- * @returns {Boolean}
  */
-function isReportApproved(report) {
-    return report && report.stateNum === CONST.REPORT.STATE_NUM.SUBMITTED && report.statusNum === CONST.REPORT.STATUS.APPROVED;
+function isReportApproved(report: OnyxEntry<Report> | undefined): boolean {
+    return report?.stateNum === CONST.REPORT.STATE_NUM.SUBMITTED && report.statusNum === CONST.REPORT.STATUS.APPROVED;
 }
 
 /**
  * Given a collection of reports returns them sorted by last read
- *
- * @param {Object} reports
- * @returns {Array}
  */
-function sortReportsByLastRead(reports) {
-    return _.chain(reports)
-        .toArray()
-        .filter((report) => report && report.reportID && report.lastReadTime)
-        .sortBy('lastReadTime')
-        .value();
+function sortReportsByLastRead(reports: OnyxCollection<Report>): Array<OnyxEntry<Report>> {
+    return Object.values(reports ?? {})
+        .filter((report) => report?.reportID && report?.lastReadTime)
+        .sort((a, b) => {
+            const aTime = a?.lastReadTime ? parseISO(a.lastReadTime) : 0;
+            const bTime = b?.lastReadTime ? parseISO(b.lastReadTime) : 0;
+            return Number(aTime) - Number(bTime);
+        });
 }
 
 /**
  * Whether the Money Request report is settled
- *
- * @param {String} reportID
- * @returns {Boolean}
  */
-function isSettled(reportID) {
+function isSettled(reportID?: string): boolean {
```

Comment:
Looks like it is a stable string variable.
```suggestion
function isSettled(reportID: string): boolean {
```

Reply:
I actually think it should be of type `string | undefined` we are using it in many screnarios like here 
<img width="371" alt="image" src="https://github.com/Expensify/App/assets/25230417/ed455c39-3c24-4e7f-9a8b-8f30b4764292">
And base on type of IOUReportID can be undefined
<img width="346" alt="image" src="https://github.com/Expensify/App/assets/25230417/3811c006-22d4-4054-9a75-394076bc251c">

Reply:
Sorry, but I haven't found any case where we pass an unpredictable value. Moreover, `isSettled` itself uses `reportID` directly without any pre-checks.

<img width="647" alt="Screenshot 2023-10-25 at 11 23 44" src="https://github.com/Expensify/App/assets/57314631/a55174be-6209-4ccd-a20b-594a9d3cc40b">

Reply:
@kubabutkiewicz, I've pulled the latest `main`, but still not here. Anyway, should it look like: `reportID: string | undefined` instead of `reportID?: string`?

<img width="936" alt="Screenshot 2023-10-25 at 14 39 06" src="https://github.com/Expensify/App/assets/57314631/1ea62854-8483-4216-afc5-dc71c6217b83">